### PR TITLE
Changed uiModes Button to Flexbox

### PR DESCRIPTION
**Pull request by tridip1931**
@kepta Can you review this.

### CODE REVIEW

File: css/80_app.css [hunk 1]
```diff
@@ -277,6 +282,24 @@ table.tags, table.tags td, table.tags th {
...
+------------------------------------------------------- */
+
+.w0 { width:04.1666%; }
+.w1 { width:08.3333%; min-width: 70px;}
+.w2 { width:16.6666%; }
+.w3 { width:25.0000%; }
+.w4 { width:33.3333%; }
+.w5 { width:41.6666%; }
+.w6 { width:50.0000%; }
+.w7 { width:58.3333%; }
```
Comment (kepta) [inline css/80_app.css]: Do we need the `.wX` ? I might be wrong but can't we substitute this with `flex-grow` / `flex-shrink`?

Comment (tridip1931) [reply css/80_app.css]: I believe flex-grow/flex-shrink only defines how much space to take in a given estate. 
CSS grid helps a lot if we are using lots of widths
I thought we needed width minus the `display: block`  but right now I am not using them anywhere. We can delete it.  @kepta

Comment (bhousel) [reply css/80_app.css]: I agree we might not need this, but I think the idea is for places where you have several joined things, (like buttons) you want them to try to share equal width.

e.g. for OK/Cancel, they'd have 50% width each, even though those words have very different natural width.  

I've found I still need to set flex items with a width in order to get them to layout correctly.  `flexbox-grow` is about how to distribute extra width to contained items, and `flexbox-shrink` is about how to steal width from them (I think - I might be wrong too).

Comment (kepta) [reply css/80_app.css]: Yup, but I feel we can simulate the case of proportions `%` with flex.
Suppose we want 2 guys to share `50:50`, we can use simply use default flex.
if `75:25`, flex-grow:3 and other to 1.

@tridip1931  @bhousel Is there a situation you guys can think where `width%` can't be replaced ?

Comment (tridip1931) [reply css/80_app.css]: Yes @bhousel I also faced this issue while working in flexbox for keeping some widths, upon looking for answers in google, I found this article very helpful. 
https://css-tricks.com/things-ive-learned-css-grid-layout/

TL;DR We should not depend entirely on flexbox for layout purposes. 

How do you feel about this guys? @bhousel @kepta

Comment (kepta) [reply css/80_app.css]: @tridip1931 `css-grid` is amazing, but we can't use it yet because of very low [browser support](http://caniuse.com/css-grid).

Comment (tridip1931) [reply css/80_app.css]: Oh ya. I will delete those widths. Also will figure out the flex-grow/shrink if I come across. 
Do we use flex-grow anywhere in our code right now?

Comment (bhousel) [reply css/80_app.css]: > Do we use flex-grow anywhere in our code right now?

It's part of the shorthand `flex:` property.  The keyboard shortcuts screen is probably the place I've most recently used it:

https://github.com/openstreetmap/iD/blob/d9b3b7b577ce9af3bff3eb5f6220a4685dbd7688/css/80_app.css#L3419-L3432

Comment (tridip1931) [reply css/80_app.css]: Thanks @bhousel . Thats very helpful. 👍

**Comment by bhousel** [comment]
closing as stale
